skybar/src/main/java/org/wtf/skybar/transform/TryCatchMethodVisitor.java

```diff
@@ -0,0 +1,195 @@
+        this.localVariablesSorter = localVariablesSorter;
+    }
+
+    @Override
+    protected void onMethodEnter() {
+
+        getNodesOfType(LineNumberNode.class).forEach(node -> {
+            int lineNumberLocal = localVariablesSorter.newLocal(Type.INT_TYPE);
+            lineNumberLocals.put(node.line, lineNumberLocal);
+            superVisitInsn(Opcodes.ICONST_0);
+            mv.visitVarInsn(Opcodes.ISTORE, lineNumberLocal);
+        });
+        tryStart = new Label();
+        tryEnd = new Label();
+        catchHandler = new Label();
+        mv.visitLabel(tryStart);
+        entered = true;
+    }
+
+
+    @Override
+    protected void onLineNumber(int lineNumber) {
+        SkybarRegistry.registry.registerLine(sourceFile, lineNumber);
+        if(entered) {
+            mv.visitIincInsn(lineNumberLocals.get(lineNumber), 1);
+        } else {
+            if(useInvokeDynamic()) {
+                MethodType mt = MethodType.methodType(CallSite.class, MethodHandles.Lookup.class, String.class, MethodType.class, String.class, int.class);
+
+                Handle bootstrap = new Handle(Opcodes.H_INVOKESTATIC, Type.getInternalName(SkybarRegistry.class), "bootstrap",
+                        mt.toMethodDescriptorString());
+                mv.visitInvokeDynamicInsn("visitLine", "()V", bootstrap, sourceFile, lineNumber);
+            } else {
+                mv.visitFieldInsn(GETSTATIC, "org/wtf/skybar/registry/SkybarRegistry", "registry", "Lorg/wtf/skybar/registry/SkybarRegistry;");
+                mv.visitLdcInsn(sourceFile);
+                mv.visitLdcInsn(lineNumber);
+                mv.visitMethodInsn(INVOKEVIRTUAL, "org/wtf/skybar/registry/SkybarRegistry", "getAdderForLine", "(Ljava/lang/String;I)Ljava/util/concurrent/atomic/LongAdder;", false);
+                mv.visitInsn(LCONST_1);
+                mv.visitMethodInsn(INVOKEVIRTUAL, "java/util/concurrent/atomic/LongAdder", "add", "(J)V", false);
+
+            }
+        }
+    }
+
+
+    @Override
+    public void visitFrame(int type, int nLocal, Object[] local, int nStack, Object[] stack) {
+        super.visitFrame(type, nLocal, local, nStack, stack);
+    }
+
+    @Override
+    protected void onMethodExit(int opcode) {
+        reportLinesExecuted();
+    }
+
+    private void reportLinesExecuted() {
+        getNodesOfType(LineNumberNode.class).forEach(node -> {
+            if (useInvokeDynamic()) {
+                MethodType mt = MethodType.methodType(CallSite.class, MethodHandles.Lookup.class, String.class, MethodType.class, String.class, int.class);
+
+                Handle bootstrap = new Handle(Opcodes.H_INVOKESTATIC, Type.getInternalName(SkybarRegistry.class), "bootstrapMulti",
+                        mt.toMethodDescriptorString());
+                mv.visitVarInsn(ILOAD, lineNumberLocals.get(node.line));
+                mv.visitInsn(I2L);
+                mv.visitInvokeDynamicInsn("visitLine", "(J)V", bootstrap, sourceFile, node.line);
```

Comment:
Added some comments. Unsure how much they help, kind of dense code to explain :-)